tty/usbacm/usbacm.c

```diff
@@ -196,26 +196,56 @@ static usbacm_dev_t *usbacm_get(int id)
 }
 
 
-static void _usbacm_put(usbacm_dev_t *dev)
+static void usbacm_remove(usbacm_dev_t *dev)
+{
+	fprintf(stdout, "usbacm: Device removed: %s\n", dev->path);
+	remove(dev->path);
+	if (dev->fifo != NULL) {
+		free(dev->fifo);
+		dev->fifo = NULL;
+		resourceDestroy(dev->rxLock);
+		resourceDestroy(dev->rxCond);
+	}
+	free(dev);
+}
+
+
+static void usbacm_remove_all(usbacm_dev_t *devices)
+{
+	usbacm_dev_t *next, *dev = devices;
+
+	if (dev == NULL) {
+		return;
+	}
+
+	do {
+		next = dev->next;
+		usbacm_remove(dev);
+		dev = next;
+	} while (dev != devices);
+}
+
+
+static usbacm_dev_t *_usbacm_put(usbacm_dev_t *dev)
```

Comment:
Just a suggestion: I think we can also return simply number of references left. We don't really need a pointer as a retval here.